erpnext/selling/doctype/sales_order/sales_order.js

```diff
@@ -98,10 +98,23 @@ frappe.ui.form.on("Sales Order", {
 	},
 
 	delivery_date: function(frm) {
+		if (!frm.doc.customer) {
+			frappe.throw(__('Please select a customer'));
+		}
 		$.each(frm.doc.items || [], function(i, d) {
 			d.delivery_date = frm.doc.delivery_date;
 		});
 		refresh_field("items");
+		frappe.db.get_value("Customer", { "name": frm.doc.customer }, "delivery_days", (r) => {
+			if (r.delivery_days) {
+				let day = moment(frm.doc.delivery_date).format('dddd');
```

Comment:
Should this get a check similar to the one you did in Order Desk? Only do it if `frm.doc.delivery_date` is set?